source/custom/nidaqmx_service.custom.cpp

```diff
@@ -203,27 +197,12 @@ ::grpc::Status NiDAQmxService::ReadAnalogWaveforms(::grpc::ServerContext* contex
       
       auto* y_data = waveform->mutable_y_data();
       y_data->Reserve(samples_per_chan_read);
-      for (int32 j = 0; j < samples_per_chan_read; ++j) {
-        y_data->Add(read_arrays[i][j]);
-      }
+      y_data->Add(read_arrays[i].data(), read_arrays[i].data() + samples_per_chan_read);
```

Comment:
Add() with an iterator range should grow the array up front, so you don't need Reserve() anymore.